### ISSUE TITLE
Cells can reference a shared string index that does not exist?

### ISSUE THREAD

**Issue by kherge** [kherge-php/excel#1]
I have a very large workbook, around 2.5 million rows spread across 4 worksheets.

In the very first sheet, column `B2` references a shared string (`15388`) that does not exist. In Excel, you can see the value (a GUID), but I'm not sure how this is being resolved.

```xml
<c r="B1" s="1" t="s">
  <v>15388</v>
</c>
```

The shared strings file only has `33029` shared strings.

```xml
<sst xmlns="http://schemas.openxmlformats.org/spreadsheetml/2006/main" count="6934163" uniqueCount="33029">
```

I need to figure out how shared strings are really resolved because the original "ref to index" assumption does not appear to be correct.

**Comment by kherge** [comment]
I must be tired. `15,388` out of `33,029` is perfectly valid.

The problem is how the `SharedStringsReader` handles empty `<t/>` elements. [The current design assumes that no shared string can be `null`.](https://github.com/kherge-php/excel/blob/0135c057098bb4c34a5ca40eeeccc2f86a5c06b8/src/KHerGe/Excel/Reader/SharedStringsReader.php#L129) This is now proven to be not true. The reader needs to be modified so that the iterator's state becomes invalid when the end of the document has been reached.

**Comment by kherge** [comment]
This issue has been fixed by commit b17f2fc18698ab937ba0655020ec38d90e8b2dad. Null shared strings are now handled properly.
